src/simulator.ts

```diff
@@ -295,11 +295,12 @@ export class Simulator {
     const stringify = Utility.getConfig<boolean>(
       Constants.IoTHubD2CMessageStringifyKey,
     );
-    let msg = new Message(stringify ? JSON.stringify(message) : message);
-    if (stringify) {
-      msg.contentType = "application/json";
-      msg.contentEncoding = "utf-8";
-    }
+    const msg = new Message(stringify ? JSON.stringify(message) : message);
+    // default to utf-8 encoding
+    msg.contentEncoding = "utf-8";
+    // msg.contentType must either be `undefined` or `application/json`
+    msg.contentType = "application/json";
```

Comment:
Now we always set `contentEncoding` and `contentType` - but `stringify` only controls whether the message is sent as original json string or whether it's further "stringified":
![image](https://user-images.githubusercontent.com/13545962/159798194-fd690fa1-22e7-4d0d-b018-f09c787a3fbe.png)

Reply:
I'm curious, if content type is always application/json, what is the value of the config option leading to an additional stringify?

Reply:
It's a good question. The feature existed before - and was causing issues because it was only marking the content type / encoding if we `enable` stringify - which seems counterintuitive for sending actual JSON.

To fix the logic here, since the content-type model is either `undefined` or `application/json` - I figured we'd always send the `application/json` content-type by default - but if we `stringify` the JSON - perhaps we can just send the content-type as `undefined` if it's a string (IMO it *should* be something like `text/plaintext`) and `application/json` if it's not?

Reply:
yea I like the proposed improvement (and agree with your assessment on what is more appropriate)...is the choice of undefined or application/json an SDK limitation? Service wise I'm unaware of any blockers preventing a different content type value being ingested (though there may be issues with routing) -- small test with the IoT CLI simulate + monitor-events

![image](https://user-images.githubusercontent.com/28658112/160469303-56dabe87-3b20-47a9-96fc-685c5600c8da.png)

Reply:
Yeah I believe this content-type is a limitation on the typescript model generated with the IoT device SDK:
https://github.com/Azure/azure-iot-sdk-node/blob/9f4a50203f191e184764601e8f7dacfbd4580b30/common/core/src/message.ts#L53